src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+        }}
+        onError={() => {
+          Logger.error(TAG, 'Error loading Nft preview image')
+        }}
+        resizeMode={FastImage.resizeMode.cover}
+        onLoadEnd={() => {
+          setLoading(false)
+        }}
+      >
+        {loading && <ThumbnailImagePlaceholder />}
+      </FastImage>
+    </Touchable>
+  )
+}
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState((nfts && nfts[0]) ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(360)
+  const { t } = useTranslation()
+
+  const android_ripple = {
+    color: colors.gray2,
+    foreground: true,
+    borderless: true,
+  }
+
+  // Some components that require parent state defined
+  const PlatformSpecificTopBarButtons = () => {
+    return (
+      <View style={styles.topBarButtonsContainerWrapper}>
+        <View style={styles.topBarButtonsContainer}>
+          {Platform.OS === 'ios' && (
+            <TopBarIconButton
+              onPress={() => navigateBack()}
+              icon={<BackChevronStatic />}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            />
+          )}
+          {Platform.OS === 'android' && (
+            <Pressable
+              android_ripple={android_ripple}
+              onPress={() => navigateBack()}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            >
+              <BackChevronStatic />
+            </Pressable>
+          )}
+        </View>
+      </View>
+    )
+  }
+
+  const MainImagePlaceholder = () => {
+    return (
+      <SkeletonPlaceholder
+        borderRadius={8}
+        backgroundColor={colors.gray2}
+        highlightColor={colors.white}
+      >
+        <View
+          style={{
+            height: scaledHeight,
+            width: variables.width,
+            zIndex: -1,
+          }}
+        />
+      </SkeletonPlaceholder>
+    )
+  }
+
+  const NftImageCarousel = () => {
+    return (
+      <View style={styles.nftImageCarouselContainer}>
+        <ScrollView
+          showsHorizontalScrollIndicator={false}
+          horizontal={true}
+          contentContainerStyle={styles.carouselScrollViewContentContainer}
+          style={styles.nftImageCarousel}
+          testID="NftsInfoCarousel/NftImageCarousel"
+        >
+          {/* Only display Nfts with metadata */}
+          {nfts
+            .filter((nft) => nft.metadata)
+            .map((nft) => {
+              return (
+                <View
+                  key={`${nft.contractAddress}-${nft.tokenId}`}
+                  style={styles.nftPreviewImageSharedContainer}
+                  testID={`NftsInfoCarousel/NftThumbnail/${nft.contractAddress}-${nft.tokenId}`}
+                >
+                  <NftThumbnail nft={nft} setActiveNft={setActiveNft} activeNft={activeNft} />
```

Comment:
nit: i think it's nice for components to know as little as possible. `NftThumbnail` doesn't really need to know which nft is active, it just needs to know if it should display itself as active or not. so perhaps a nicer props interface is `isActive` and `onPress` rather than `activeNft / setActiveNft`